src/transaction.ts

```diff
@@ -200,13 +226,22 @@ export class Transaction {
    *
    * @param signedBy The address of the future signer
    */
-  serializeForSigning(signedBy: IAddress): Buffer {
+  serializeForSigning(): Buffer {
     // TODO: for appropriate tx.version, interpret tx.options accordingly and sign using the content / data hash
-    let plain = this.toPlainObject(signedBy);
+    let plain = this.toPlainObject();
     // Make sure we never sign the transaction with another signature set up (useful when using the same method for verification)
     if (plain.signature) {
       delete plain.signature;
     }
+
+    if (plain.guardianSignature) {
+      delete plain.guardianSignature;
+    }
+
+    if ((plain.guardian == undefined) || (plain.guardian == "")) {
```

Comment:
We should use `===`, because... Javascript. But I think any falsy value should delete the property, so you could use `if (!plain.guardian)`

Reply:
changed